testsuite/pytests/test_eprop_plasticity.py

```diff
@@ -0,0 +1,738 @@
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Test functionality of e-prop plasticity.
+"""
+
+import nest
+import numpy as np
+import pytest
+
+
+def test_ConnectNeuronsWithEpropSynapse():
+    """Ensures that the restriction to supported neuron models works."""
+
+    nest.set_verbosity("M_WARNING")
+
+    supported_source_models = [
+        "eprop_iaf_psc_delta",
+        "eprop_iaf_psc_delta_adapt",
+    ]
+    supported_target_models = supported_source_models + ["eprop_readout"]
+
+    # Connect supported models with e-prop synapse
+
+    for nms in supported_source_models:
+        for nmt in supported_target_models:
+            nest.ResetKernel()
+
+            ns = nest.Create(nms, 1)
+            nt = nest.Create(nmt, 1)
+
+            nest.Connect(ns, nt, {"rule": "all_to_all"}, {"synapse_model": "eprop_synapse"})
+
+    # Ensure that connecting not supported models fails
+
+    for nm in [n for n in nest.node_models if n not in supported_target_models]:
+        nest.ResetKernel()
+
+        n = nest.Create(nm, 2)
+
+        with pytest.raises(nest.kernel.NESTError):
+            nest.Connect(n, n, {"rule": "all_to_all"}, {"synapse_model": "eprop_synapse"})
+
+
+def test_EpropRegression():
+    """
+    Test correct computation of weights for a regresion task  by comparing the simulated losses with
+    losses obtained in a simulation with the original, verified NEST implementation and with the original
+    TensorFlow implementation.
+    """
+
+    # Initialize random generator
+
+    rng_seed = 1
+    np.random.seed(rng_seed)
+
+    # Define timing of task
+
+    n_batch = 1
+    n_iter = 5
+
+    steps = {
+        "recall": 1,
+        "recall_onset": 0,
+        "sequence": 100,
+        "shift": 4,
+    }
+
+    steps["task"] = n_iter * n_batch * steps["sequence"]
+    steps["sim"] = steps["task"] + steps["shift"]
+
+    duration = {"step": 1.0}
+
+    duration.update({key: value * duration["step"] for key, value in steps.items()})
+
+    # Set up simulation
+
+    params_setup = {
+        "eprop_reset_neurons_on_update": True,
+        "eprop_update_interval": duration["sequence"],
+        "print_time": False,
+        "resolution": duration["step"],
+        "total_num_virtual_procs": 1,
+    }
+
+    nest.ResetKernel()
+    nest.set(**params_setup)
+
+    # Create neurons
+
+    n_in = 10
+    n_rec = 10
+    n_out = 1
+
+    params_nrn_rec = {
+        "C_m": 1.0,
+        "c_reg": 300.0,
+        "gamma": 0.3,
+        "E_L": 0.0,
+        "f_target": 10.0,
+        "I_e": 0.0,
+        "propagator_idx": 0,
+        "surrogate_gradient": "piecewise_linear",
+        "t_ref": 0.0,
+        "tau_m": 30.0,
+        "V_m": 0.0,
+        "V_th": 0.03,
+    }
+
+    params_nrn_out = {
+        "C_m": 1.0,
+        "E_L": 0.0,
+        "I_e": 0.0,
+        "loss": "mean_squared_error",
+        "start_learning": duration["recall_onset"],
```

Comment:
This parameter exists no longer.

Reply:
Fixed in https://github.com/nest/nest-simulator/pull/2867/commits/79b0ecaf778214cdc4f4af10e62b83307ff15035